tests/api/route-services/work-response.test.js

```diff
@@ -270,103 +216,86 @@ describe('tests for the work-response service', () => {
-        timeout: '2020-08-17T08:17:52.026Z',
-        body: {
-          name: 'DifferentialExpression',
-          cellSet: 'louvain-0',
-          compareWith: 'louvain-1',
-        },
+        ...responseFormat.request,
         pagination,
       },
-      response: {
-        cacheable: true,
-        error: false,
-      },
       results: [
         {
           'content-type': 'application/json',
           'content-encoding': 'utf-8',
-          body: '{"rows": [{"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "A"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "B"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "C"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "D"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "E"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "F"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "G"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "H"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "I"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "J"}]}',
+          body: RESPONSE_BODY,
           type: 'inline',
         },
       ],
     };
 
-    const w = new WorkResponseService(io, workResponse);
-    const expectedResponse = JSON.parse(JSON.stringify(workResponse));
-    expectedResponse.results
-      .forEach((res) => delete res.type);
-    expectedResponse.results[0].body = 'my amazing body';
+    client.on(`WorkResponse-${workResponse.request.uuid}`, (res) => {
+      const expectedResponse = JSON.parse(JSON.stringify(workResponse));
+      // eslint-disable-next-line no-param-reassign
+      expectedResponse.results.forEach((result) => delete result.type);
 
-    w.handleResponse().then(() => {
-      expect(handlePagination).toHaveBeenCalledTimes(1);
-      expect(handlePagination).toHaveBeenCalledWith([
-        {
-          'content-type': 'application/json',
-          'content-encoding': 'utf-8',
-          body: '{"rows": [{"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "A"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "B"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "C"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "D"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "E"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "F"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "G"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "H"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "I"}, {"pval": 0.2, "qval": 0.3, "log2fc": 2.4, "gene_names": "J"}]}',
-        },
-      ], pagination);
-      return done();
+      const expectedBody = JSON.parse(RESPONSE_BODY);
+      expectedBody.total = expectedBody.rows.length;
+      expectedBody.rows = expectedBody.rows.slice(0, pagination.limit);
+      expectedResponse.results[0].body = JSON.stringify(expectedBody);
+
+      expect(res).toEqual(expectedResponse);
+      done();
```

Comment:
In the old tests handlePagination was mocked out so it wasn't really testing anything. This is now fixed.